pkg/build/builder/cmd/scmauth/password.go

```diff
@@ -86,7 +87,12 @@ func (u UsernamePassword) Setup(baseDir string) error {
 	fmt.Fprintf(gitconfig, UserPassGitConfig, gitcredentials.Name())
 	fmt.Fprintf(gitcredentials, "%s", usernamePasswordURL.String())
 
-	return ensureGitConfigIncludes(gitconfig.Name())
+	sourceURL := u.SourceURL
+	sourceURL.User = nil
```

Comment:
don't we have to leave the username, and strip the password? From the git docs: "If the protocol, hostname, and username (if we already have one) match..."